From bffd615cad5486b1e4613506e48ab0bcf41d19bc Mon Sep 17 00:00:00 2001
From: Deepanjan Roy <dproy@chromium.org>
Date: Wed, 26 May 2021 14:19:40 -0400
Subject: [PATCH] Reland "Convert generate_about_tracing target to py3"

This is a reland of b84752766a4161ab1d3e117625fcbaedc4ae07c5

Original change's description:
> Convert generate_about_tracing target to py3
>
> Need an additional str.encode to make python3 happy. Checked that all
> the code still works under python2 as well.
>
> Bug: chromium:1205617
> Change-Id: I4217ed4213b57209a396f604396923985d1ac2d2
> Reviewed-on: https://chromium-review.googlesource.com/c/catapult/+/2920487
> Commit-Queue: Deep Roy <dproy@chromium.org>
> Commit-Queue: John Chen <johnchen@chromium.org>
> Auto-Submit: Deep Roy <dproy@chromium.org>
> Reviewed-by: John Chen <johnchen@chromium.org>

Bug: chromium:1205617
Change-Id: I57b4c55afb454944d2d729322d9af19995d0514a
Reviewed-on: https://chromium-review.googlesource.com/c/catapult/+/2924526
Auto-Submit: John Chen <johnchen@chromium.org>
Commit-Queue: Deep Roy <dproy@chromium.org>
Reviewed-by: Deep Roy <dproy@chromium.org>
---
 src/third_party/catapult/BUILD.gn                            | 3 +++
 .../catapult/common/py_vulcanize/py_vulcanize/generate.py    | 2 +-
 .../common/py_vulcanize/py_vulcanize/parse_html_deps.py      | 5 +++++
 src/third_party/catapult/tracing/BUILD.gn                    | 4 +---
 4 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/src/third_party/catapult/BUILD.gn b/src/third_party/catapult/BUILD.gn
index 0e10ca22c1..b228edbed9 100644
--- a/src/third_party/catapult/BUILD.gn
+++ b/src/third_party/catapult/BUILD.gn
@@ -56,6 +56,7 @@ group("telemetry_chrome_test_support") {
     "third_party/WebOb/",
     "third_party/apiclient/",
     "third_party/beautifulsoup4/",
+    "third_party/beautifulsoup4-4.9.3/",
     "third_party/cachetools/",
     "third_party/certifi/",
     "third_party/chai/",
@@ -68,6 +69,7 @@ group("telemetry_chrome_test_support") {
     "third_party/gae_ts_mon/",
     "third_party/google-auth/",
     "third_party/graphy/",
+    "third_party/html5lib-1.1/",
     "third_party/html5lib-python/",
     "third_party/httplib2/",
     "third_party/idb/",
@@ -100,6 +102,7 @@ group("telemetry_chrome_test_support") {
     "third_party/uritemplate/",
     "third_party/urllib3/",
     "third_party/webapp2/",
+    "third_party/webencodings-0.5.1/",
     "third_party/webtest/",
   ]
 
diff --git a/src/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py b/src/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
index 8af373102f..c5c1ef7709 100644
--- a/src/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
+++ b/src/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
@@ -208,7 +208,7 @@ def _MinifyCSS(css_text):
                          stdin=subprocess.PIPE,
                          stdout=subprocess.PIPE,
                          stderr=subprocess.PIPE)
-    res = p.communicate(input=css_text)
+    res = p.communicate(input=css_text.encode('utf-8'))
     errorcode = p.wait()
     if errorcode != 0:
       sys.stderr.write('rCSSmin exited with error code %d' % errorcode)
diff --git a/src/third_party/catapult/common/py_vulcanize/py_vulcanize/parse_html_deps.py b/src/third_party/catapult/common/py_vulcanize/py_vulcanize/parse_html_deps.py
index 88ce21820c..fc9f45199f 100644
--- a/src/third_party/catapult/common/py_vulcanize/py_vulcanize/parse_html_deps.py
+++ b/src/third_party/catapult/common/py_vulcanize/py_vulcanize/parse_html_deps.py
@@ -31,6 +31,11 @@ def _InitBeautifulSoup():
   html5lib_path = os.path.join(catapult_path, 'third_party', 'html5lib-python')
   _AddToPathIfNeeded(html5lib_path)
 
+  if six.PY3:
+    webencodings_path = os.path.join(
+        catapult_path, 'third_party', 'webencodings-0.5.1')
+    _AddToPathIfNeeded(webencodings_path)
+
   six_path = os.path.join(catapult_path, 'third_party', 'six')
   _AddToPathIfNeeded(six_path)
 
diff --git a/src/third_party/catapult/tracing/BUILD.gn b/src/third_party/catapult/tracing/BUILD.gn
index 8026066a1f..889d30bc72 100644
--- a/src/third_party/catapult/tracing/BUILD.gn
+++ b/src/third_party/catapult/tracing/BUILD.gn
@@ -2,15 +2,13 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//build/config/python.gni")
 import("trace_viewer.gni")
 
 # TODO: ideally this would go into the target_gen_dir, but this requires some
 # changes to the scripts that process them.
 output_resource_dir = "$root_gen_dir/content/browser/tracing"
 
-# TODO(crbug.com/1112471): Make this work under Python 3.
-python2_action("generate_about_tracing") {
+action("generate_about_tracing") {
   script = "bin/generate_about_tracing_contents"
 
   inputs = tracing_files + tracing_python_files
